presto-hive/src/main/java/io/prestosql/plugin/hive/HivePageSourceProvider.java

```diff
@@ -162,7 +166,27 @@ public ConnectorPageSource createPageSource(ConnectorTransactionHandle transacti
                 path,
                 bucketNumber,
                 fileSize,
-                fileModifiedTime);
+                fileModifiedTime,
+                hiveStorageTimeZone);
+
+        // Check if prefilled values match dynamic filter
+        Map<HiveColumnHandle, Domain> dynamicFilterDomains = dynamicFilter.getDomains().get();
+        if (!dynamicFilterDomains.isEmpty()) {
+            List<ColumnMapping> prefilledColumnMappings = ColumnMapping.extractPrefilledColumnMappings(columnMappings);
```

Comment:
this performs partition pruning on node using `dynamicFilter`. I think it's fine, but it will also be addressed by cluster wide dynamic filtering that is in progress.

I would make the mechanism more generic though, so that it works on intersection of `dynamicFilter` and `effectivePredicate`

Reply:
So I think if the cluster-wide dynamic filtering will be checked in soon, we don't need to check in this PR. Because it brings no benefits but overhead. I'd like to take time learning the current status of cluster-wide dynamic filtering work before I go ahead addressing this PR's comments. Please LMK if you have other advice.

Reply:
I am actively working on https://github.com/prestosql/presto/pull/1072/ 
While the feature itself works, I'm currently working on addressing comments about the way in which dynamic filters are passed to the coordinator. I'm hoping to finish this soon.

Reply:
Get it. Thank you for the update! @raunaqmorarka